### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by XkunW**
# PR Type
[Feature]

# Short Description

- Created `vec-inf` CLI and uploaded to PyPI
- Added support for Llama-3.1, gemma-2, phi-3 families and other model variants

# Tests Added
N/A

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -54,19 +54,19 @@ RUN python3.10 -m pip install --upgrade pip
 RUN python3.10 -m pip install poetry
 
 # Clone the repository
-RUN git clone https://github.com/VectorInstitute/vector-inference /vec-inf
+RUN git clone -b develop https://github.com/VectorInstitute/vector-inference /vec-inf
```
Comment (jwilles) [inline Dockerfile:57]: Why are we cloning here? Is there a reason we build from source? If so, we should have something more robust than develop. If not, we should simply install the pip package.

Comment (XkunW) [reply Dockerfile:57]: Good catch, forgot to update this

File: vec_inf/cli/_cli.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+    "--log-dir",
+    type=str,
+    help='Path to slurm log directory'
+)
+@click.option(
+    "--json-mode",
+    is_flag=True,
+    help='Output in JSON string',
+)
+def launch(
```
Comment (jwilles) [inline vec_inf/cli/_cli.py:87]: Is it still possible for someone to run their own model weights? Assuming it conforms to an architecture supported by vllm? Also, what about passing in additional config for launch? Max token length ect. Is that still possible?

Comment (XkunW) [reply vec_inf/cli/_cli.py:87]: If the architecture is supported by vllm, then yes, as long as they supply values for all the optional arguments. W.r.t max tokens, that is set in chat/completions API endpoint, not in model launch.

Comment (jwilles) [reply vec_inf/cli/_cli.py:87]: Sorry not max tokens, I meant max_model_len. Can these be set via the launch command?

**Comment by jwilles** [comment]
Doesn't need to happen in this PR but can we fix the checks that are failing?

File: vec_inf/cli/_cli.py (hunk 2 above)
Comment (XkunW) [reply vec_inf/cli/_cli.py:87]: Yes that's an optional argument

**Comment by XkunW** [comment]
> Doesn't need to happen in this PR but can we fix the checks that are failing?

Lemme see if I can quickly fix that, I've tried to previously but didn't quite finish it

**Pull request merged**